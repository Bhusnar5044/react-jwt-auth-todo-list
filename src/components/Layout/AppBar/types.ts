export type Props = {
    // Header prop type
    pages: { label: string; path: string }[];
};
