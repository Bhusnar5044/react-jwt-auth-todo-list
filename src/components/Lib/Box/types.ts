export type Props = {
    display?: 'flex' | 'inline-flex';
    flexDirection?: 'row' | 'column';
    justifyContent?: 'center' | 'space-between' | 'space-around';
    alignItems?: 'center' | 'start' | 'end';
    flexWrap?: 'wrap';
    gap?: string;
    margin?: string;
};
