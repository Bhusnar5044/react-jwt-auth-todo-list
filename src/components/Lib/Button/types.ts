export type Props = {
    type?: 'button' | 'submit' | 'reset';
    disabled?: boolean;
    onClick?: () => void;
    loading?: boolean;
};
