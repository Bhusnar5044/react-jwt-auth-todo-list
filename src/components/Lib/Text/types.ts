import { FontVariants, FontWeights } from '@theme/font/types';
import { HTMLProps } from '@utils/types';

interface CommonProps extends HTMLProps<HTMLElement> {
    textColor?: string;
    uppercase?: boolean;
    lineThrough?: boolean;
    fullWidth?: boolean;
    as?: keyof JSX.IntrinsicElements | React.ComponentType<any>;
}

export interface TextProps extends CommonProps {
    textVariant?: FontVariants;
    textWeight?: FontWeights;
    textAlign?: 'initial' | 'left' | 'right' | 'center';
}
