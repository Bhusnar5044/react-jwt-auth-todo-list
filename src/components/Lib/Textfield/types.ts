import { ChangeEvent } from 'react';

export interface InputWrapperProps {
    fullWidth?: boolean;
    minWidth?: string;
}
export interface Props extends InputWrapperProps {
    label: string;
    name?: string;
    value: any;
    type?: string;
    onChange: (e: ChangeEvent<HTMLInputElement>) => void;
    placeholder?: string;
}
