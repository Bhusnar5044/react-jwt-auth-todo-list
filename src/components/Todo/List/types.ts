export type Task = { text: string; done: boolean };

export type Props = {
    tasks: Task[];
    setTasks: React.Dispatch<React.SetStateAction<Task[]>>;
};
