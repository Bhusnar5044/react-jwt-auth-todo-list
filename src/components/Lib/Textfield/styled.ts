import { theme } from '@theme';
import { styled } from '@utils';
import { InputWrapperProps } from './types';

export const Label = styled.label`
    position: absolute;
    font-size: 1rem;
    left: 0;
    top: 50%;
    /* transform: translateY(-50%); */
    background-color: transparent;
    color: ${theme.colors.info.dark};
    padding: 0 0.3rem;
    margin: 0 0.5rem;
    transition: 0.1s ease-out;
    transform-origin: left top;
    pointer-events: none;
`;

export const Input = styled.input`
    font-size: 1rem;
    outline: none;
    border: 1px solid ${theme.colors.info.light};
    border-radius: 5px;
    padding: 0.5rem 0.7rem;
    color: ${theme.colors.info.light};
    transition: 0.1s ease-out;
    box-sizing: border-box;

    &:focus {
        border-color: #6200ee;
    }

    &:focus + label {
        color: #6200ee;
        top: 0;
        transform: translateY(-50%) scale(0.9);
    }

    &:not(:placeholder-shown) + label {
        top: 0;
        transform: translateY(-50%) scale(0.9);
    }
`;

export const InputWrapper = styled.div<InputWrapperProps>`
    position: relative;
    width: ${({ fullWidth }) => fullWidth && '100%'};
    min-width: ${({ minWidth }) => minWidth};

    ${Input} {
        width: ${({ fullWidth }) => fullWidth && '100%'};
        min-width: ${({ minWidth }) => minWidth};
    }
`;
