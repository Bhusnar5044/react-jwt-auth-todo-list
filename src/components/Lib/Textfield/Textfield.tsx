import { FC, forwardRef, memo } from 'react';
import { Input, InputWrapper, Label } from './styled';
import { Props } from './types';

export const Textfield: FC<Props> = memo(
    forwardRef<HTMLInputElement, Props>(({ label, minWidth, fullWidth, ...props }, ref) => {
        return (
            // <InputWrapper {...{ minWidth, fullWidth }}>
            //     <Input ref={ref} type="text" {...props} />
            //     <Label>{label}</Label>
            // </InputWrapper>
            <InputWrapper for="inp" {...{ minWidth, fullWidth }}>
                <Input ref={ref} type="text" id="inp" placeholder="&nbsp;" {...props} />
                <Label>{label}</Label>
                <span className="focus-bg"></span>
            </InputWrapper>
        );
    }),
);

Textfield.displayName = 'Textfield';
