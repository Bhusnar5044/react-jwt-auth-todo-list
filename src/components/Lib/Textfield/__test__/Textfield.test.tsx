import { fireEvent, render, screen } from '@test-utils';
import { Textfield } from '../Textfield';
import { Props } from '../types';

const onChange = jest.fn();
const defaultProps: Props = {
    label: 'Dummy',
    name: 'Dummy',
    value: '',
    onChange,
    placeholder: 'dummy placeholder',
};
const renderer = (props = defaultProps) => render(<Textfield {...props} />);

describe('Textfield', () => {
    it('should render properly', () => {
        const { container } = renderer();
        expect(container).toMatchSnapshot();
    });

    it('should call onChange event', () => {
        renderer();
        const Input = screen.getByPlaceholderText(/dummy placeholder/i);
        fireEvent.change(Input, { target: { value: '123' } });
        expect(onChange).toBeCalled();
    });

    it('should render textField with value', () => {
        renderer({ ...defaultProps, value: '123' });
        expect(screen.getByPlaceholderText(/dummy placeholder/i)).toHaveValue('123');
    });

    it('should disable the texField if disable prop is provided', () => {
        renderer({ ...defaultProps, disabled: true });
        expect(screen.getByPlaceholderText(/dummy placeholder/i)).toBeDisabled();
    });
});
