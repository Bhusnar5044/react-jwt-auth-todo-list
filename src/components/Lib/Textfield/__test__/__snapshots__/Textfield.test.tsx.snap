// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textfield should render properly 1`] = `
.c0 {
  position: relative;
}

.c2 {
  position: absolute;
  font-size: 1rem;
  left: 0;
  top: 50%;
  transform: translateY(-50%);
  background-color: white;
  color: gray;
  padding: 0 0.3rem;
  margin: 0 0.5rem;
  transition: 0.1s ease-out;
  transform-origin: left top;
  pointer-events: none;
}

.c1 {
  font-size: 1rem;
  outline: none;
  border: 1px solid gray;
  border-radius: 5px;
  padding: 0.5rem 0.7rem;
  color: gray;
  transition: 0.1s ease-out;
}

.c1:focus {
  border-color: #6200ee;
}

.c1:focus+label {
  color: #6200ee;
  top: 0;
  transform: translateY(-50%) scale(0.9);
}

.c1:not(:placeholder-shown)+label {
  top: 0;
  transform: translateY(-50%) scale(0.9);
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      name="Dummy"
      placeholder="dummy placeholder"
      type="text"
      value=""
    />
    <label
      class="c2"
    >
      Dummy
    </label>
  </div>
</div>
`;
